flows/events/contact_channel_changed.go

```diff
@@ -39,6 +42,40 @@ func (e *ContactChannelChangedEvent) Apply(run flows.FlowRun) error {
 		return err
 	}
 
-	run.Contact().SetChannel(channel)
+	contactURNs := run.Contact().URNs()
+
+	// change any tel URNs this contact has to use this channel (other URN types may be channel specific)
+	if channel.SupportsScheme(urns.TelScheme) {
+		for _, urn := range contactURNs {
+			if urn.URN.Scheme() == urns.TelScheme {
+				urn.SetChannel(channel)
+			}
+		}
+	}
+
+	// if our scheme isn't the highest priority
+	if len(contactURNs) > 0 && !channel.SupportsScheme(contactURNs[0].Scheme()) {
+
+		// find the highest priority supported by this channel
+		var newPreferredURN *flows.ContactURN
+		for _, urn := range contactURNs {
+			if channel.SupportsScheme(urn.Scheme()) {
+				newPreferredURN = &urn
+			}
+		}
+
+		// update the highest URN of the right scheme to be highest priority
+		if newPreferredURN != nil {
+			newURNs := make(flows.URNList, 1)
+			newURNs[0] = *newPreferredURN
+
+			for _, urn := range contactURNs {
+				if urn.URN != newPreferredURN.URN {
+					newURNs = append(newURNs, urn)
+				}
+			}
+		}
```

Comment:
Is the above working? How are `contactURNs` being updated? Seems you are creating a new list and reordering it but how does that end up updating the contact urns?

Also, can't this just be a two element swap instead of creation of a whole new list?